localstack/services/s3/s3_listener.py

```diff
@@ -666,6 +666,25 @@ def return_response(self, method, path, data, headers, response):
             response.status_code = 204
             return response
 
+        # emulate ErrorDocument functionality if a website is configured
+
+        if method == 'GET' and response.status_code == 404 and parsed.query != 'website':
+            s3_client = aws_stack.connect_to_service('s3')
+
+            try:
+                # Verify the bucket exists in the first place--if not, we want normal processing of the 404
+                s3_client.head_bucket(Bucket=bucket_name)
+                website_config = s3_client.get_bucket_website(Bucket=bucket_name)
+
+                if 'ErrorDocument' in website_config and 'Key' in website_config['ErrorDocument']:
```

Comment:
nit: perhaps a bit easier:
```
error_doc_key = website_config.get('ErrorDocument', {}).get('Key')
if error_doc_key:
    error_object = s3_client.get_object(Bucket=bucket_name, Key=error_doc_key)
    ...
```

Reply:
Makes sense. Updated in e597444c09b063d1bd5d04f85b894a1b82d7e848